### PR TITLE
2.4 - delete obs addon when enableMetrics is set to false

### PR DESCRIPTION
**Pull request by subbarao-meduri**
related: [#19015](https://github.com/stolostron/backlog/issues/19015)

Signed-off-by: Subbarao Meduri <smeduri@redhat.com>

### CODE REVIEW

File: tests/pkg/utils/mco_oba.go [hunk 1]
```diff
@@ -31,6 +31,19 @@ func CheckOBAStatus(opt TestOptions, namespace, status string) error {
...
 }
 
+func CheckOBADeleted(opt TestOptions, namespace string) error {
+	dynClient := NewKubeClientDynamic(
+		opt.HubCluster.ClusterServerURL,
+		opt.KubeConfig,
+		opt.HubCluster.KubeContext)
+
+	_, err := dynClient.Resource(NewMCOAddonGVR()).Namespace(namespace).Get(context.TODO(), "observability-addon", metav1.GetOptions{})
+	if err == nil {
```
Comment (marcolan018) [inline tests/pkg/utils/mco_oba.go:41]: better to check error status/message, only return nil when error is notfound one

Comment (subbarao-meduri) [reply tests/pkg/utils/mco_oba.go]: The code is changed to specifically check for `NotFound` error.

**Pull request merged**